### PR TITLE
perf: Cache stats.json loaded from class path (#10303)

### LINKED ISSUE #10116: loadtesting causes stats.json to fail to load sometimes
### Description of the bug / feature
I ran some loadtests with apache jmeter. The application works fine during debugging and also during the loadtests until some threshold (around 1000 users). Above that threshold during UI Initialization the application fails to load the stats.json file (I guess due too many open files).
### Minimal reproducible example
Haven't had the time yet to create a minimal example.
### Expected behavior
Opening a ui should cause the server to open the same file each time.
### Actual behavior
It fails to open stats.json
### Versions:
    - Vaadin / Flow version: Vaadin 14.4.7 (production mode, compatibility mode disabled)
    - Java version: 8
    - OS version: debian 9
    - Browser version (if applicable): apache jmeter
    - Application Server (if applicable): tomcat 8
    - IDE (if applicable):

I tried to capture some useful messages from the logs:
```
ERROR [http-nio-8080-exec-71] (FrontendUtils:505) - Cannot get the 'stats.json' from the classpath 'META-INF/VAADIN/config/stats.json'
WARN  [http-nio-8080-exec-71] (UIInitListener:67) - Unexpected exception occured: BootstrapException: Unable to read webpack stats file.
DEBUG [http-nio-8080-exec-71] (UIInitListener:70) - Unexpected exception occured: com.vaadin.flow.server.BootstrapException: Unable to read webpack stats file.
        at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupFrameworkLibraries(BootstrapHandler.java:894)
        at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupDocumentHead(BootstrapHandler.java:771)
        at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.getBootstrapPage(BootstrapHandler.java:540)
        at com.vaadin.flow.server.BootstrapHandler.synchronizedHandleRequest(BootstrapHandler.java:481)
        at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
        at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1547)
        at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:247)
        at com.vaadin.cdi.CdiVaadinServlet.service(CdiVaadinServlet.java:67)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.openejb.server.httpd.EEFilter.doFilter(EEFilter.java:65)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.tomee.catalina.OpenEJBValve.invoke(OpenEJBValve.java:45)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
        at org.apache.tomee.catalina.OpenEJBSecurityListener$RequestCapturer.invoke(OpenEJBSecurityListener.java:97)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1627)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: The stats file from webpack (stats.json) was not found.
The application is running in production mode.Verify that build-frontend task has executed successfully and that stats.json is on the classpath.Or switch application to development mode.
        at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.appendNpmBundle(BootstrapHandler.java:926)
        at com.vaadin.flow.server.BootstrapHandler$BootstrapPageBuilder.setupFrameworkLibraries(BootstrapHandler.java:892)
        ... 38 more


```

### PR DESCRIPTION
**Pull request by mshabarov**
Caches stats.json file loaded from class path and improves the performance by not being opening too many stats.json files under the load.

Fixes #10116

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java [hunk 1]
```diff
@@ -37,22 +37,34 @@
 import com.vaadin.flow.di.Lookup;
 import com.vaadin.flow.di.ResourceProvider;
 import com.vaadin.flow.function.DeploymentConfiguration;
+import com.vaadin.flow.server.VaadinContext;
```
Comment (taefi) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java:40]:
```suggestion
```

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java [hunk 2]
```diff
@@ -37,22 +37,34 @@
...
 import com.vaadin.flow.server.MockVaadinServletService;
 import com.vaadin.flow.server.ServiceException;
-import com.vaadin.flow.server.VaadinContext;
```
Comment (taefi) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java:42]: I believe it is better to keep the import order as it was.

**Review by taefi — CHANGES_REQUESTED** [review]
Completely fine, just a trivial import order fixation...

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java (hunk 2 above)
Comment (mshabarov) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/FrontendUtilsTest.java]: Sure, thanks!

**Review by taefi — APPROVED** [review]

**Pull request merged**